wallet-new/test/RequestSpec.hs

```diff
@@ -74,6 +74,12 @@ filterSpec = do
                     `shouldSatisfy`
                         isLeft
 
+            it "supports IN" $ do
+                parseFilterOperation pw pcoin "IN[1,2,3]"
+                    `shouldBe`
+                        Right
+                            (FilterIn (map Core.mkCoin [1,2,3]))
```

Comment:
Good lad! 🎉 😁 😍 

What about other draconian cases like `IN[]` etc? Should we test for failures as well?

Reply:
mmmm, we don't currently require a `NonEmpty`, so I guess `IN[]` is just "please don't return anything to me" :joy:

Reply:
Got you! As long as it doesn't blow up in the user face, I'm happy 😁 (More concerned about the possibility that passing an empty list down the guts of the filtering machinery could cause some non-total function to go bonkers).

Reply:
I suppose we could change it to `NonEmpty`, and do an `err400` if they provide nonempty list. That might be the less surprising thing?

Reply:
Otherwise, let's destroy all non-total functions :smile: